unirest/src/test/java/BehaviorTests/GZipTest.java

```diff
@@ -26,12 +26,38 @@
 package BehaviorTests;
 
 import kong.unirest.HttpResponse;
+import kong.unirest.ObjectMapper;
 import kong.unirest.Unirest;
 import org.junit.Test;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+import java.util.Map;
 
 public class GZipTest extends BddTest {
+    @Test
+    public void emptyGzip() {
+        ObjectMapper om = Unirest.config().getObjectMapper();
+        ObjectMapper omWithNull = new ObjectMapper() {
+          @Override
+          public <T> T readValue(String value, Class<T> valueType) {
+            return value == null || value.trim().isEmpty() ? null : om.readValue(value, valueType);
+          }
+          @Override
+          public String writeValue(Object value) {
+            return om.writeValue(value);
+          }
+        };
+        Unirest.config().setObjectMapper(omWithNull);
+        HttpResponse<Map<?,?>> resp = Unirest.post(MockServer.EMPTY_GZIP)
+                .header("Content-Type", "application/json")
+                .accept( "application/json")
+                .asObject(Map.class);
+        Unirest.config().setObjectMapper(om);
+        assertFalse(resp.getParsingError().isPresent());
```

Comment:
Thank you for doing one of these tests! They often get missed. I really appreciate it.

I think  the test conflates two issues:
   1. That the GZIP method doesn't bomb on an empty array
   1. The behavior of an ObjectMapper when presented with null or empty data.

In reality, I would expect the JSON parser to make the  call  on if a null or empty response should be parsed into an empty Map, however that is a different issue I think. 

This test would be more clear if we simply do a asString() which I would expect to produce an empty String (which is what new String(new byte[0]) gives you). So this can  be simplified to: 

```
HttpResponse<String> result  = Unirest.post(MockServer.EMPTY_GZIP)
                                      .header("Content-Type", "application/json")
                                      .accept( "application/json")
                                      .asString();

        assertEquals("", result.getBody());
        assertEquals(false, result.getParsingError().isPresent());
```

Reply:
Updated the code as suggested. The old behavior was to return null, with this change it would return empty string instead of null.